src/pal/src/safecrt/strlen_s.cpp

```diff
@@ -42,7 +42,7 @@
 *
 *******************************************************************************/
 
-size_t __cdecl strnlen(const char *str, size_t maxsize)
+size_t __cdecl strnlen(const char *str, size_t maxsize) throw()
```

Comment:
I've synced your changes and generated a preprocessed version of `strlen_s.cpp` to see where the throws() comes from. And the problem is that it gets the prototype from /usr/include/string.h, where it has the `throws()` annotation. 
The proper fix is to name the strnlen as PAL_strnlen (in both the `strlen_s.cpp` and the `mbusafecrt.h`) and then add `#define strnlen PAL_strnlen` to the `pal.h` and remove `#define strnlen DUMMY_strnlen` from the `palinternal.h`